master_visu_scripts/vtk_plot.py

```diff
@@ -11,48 +11,37 @@
 from tqdm import tqdm
 from cmcrameri import cm
 
-def plot(file,field='density',bounds=None,contours=False,
+def plot(file,field,bounds,ax=None,contours=False,
          cfields=['crust_upper','crust_lower','mantle_lithosphere'],
-         null_field='asthenosphere',off_screen=True,output='img.png',
-         camera=None,plot_scalar_bar=True,**kwargs):
+         null_field='asthenosphere',plot_scalar_bar=True,**kwargs):
     """
     Plot 2D ASPECT results using Pyvista.
-    
-    Best practice is to employ this function twice, once to determine the
-    appropriate camera value with off_screen=False and camera=None, and a
-    second time with off_screen=True and the camera specified. Off_screen=
-    False triggers an interactive window where the camera can be adjusted.
-    Specifying bounds will clip the image and can help ensure that the camera
-    position is correct; some trial and error may be necessary.
 
     Parameters
     ----------
     file : VTU or PVTU file to plot
-    field : Field to use for color. The default is 'density'.
-    bounds : Bounds by which to clip the plot. The default is None.
+    field : Field to use for color.
+    bounds : Bounds by which to clip the plot.
+    ax: Matplotlib axes on which to plot results. The default is None.
     contours : Boolean for whether to add temperature contours. 
         The default is False.
     cfields : Names of compositional fields to use if field is 'comp_field.' 
         The default is ['crust_upper','crust_lower','mantle_lithosphere'].
     null_field : Null field if field is 'comp_field.'
         The default is 'asthenosphere'.
-    off_screen : Boolean for whether Pyvista plotting occurs off-screen.
-        The default is True.
-    output : Name of image file to output screenshot
-    camera : Position for Pyvista camera. Needs to be a list
-        with focal point, position, and viewup. The default is None.
 
     Returns
     -------
-    cpos: Camera position if plotting onscreen [focal point, position,viewup]
-    img : img object that can be saved or plotted.
+    ax: Matplotlib axes with results plotted.
 
     """
     
     mesh = pv.read(file)
     
-    if bounds is not None:
-        mesh = mesh.clip_box(bounds=bounds,invert=False)
+    km2m = 1000
+    bounds_m = [bound*km2m for bound in bounds] # Convert bounds to m
+    bounds_3D = bounds_m + [0,0]

```

Comment:
In `stress_plots` the bounds currently set are already for 3 dims, adding another will make `clip_box` fail. The bounds input set in `stress_plots` should be changed.

Reply:
This has been addressed by d09ccf2 and 0b41c06.